packages/zoe/zoe.js

```diff
@@ -264,87 +367,158 @@ Unimplemented installation moduleFormat ${moduleFormat}`;
+            // change in the PR to use invites rather than escrow receipts.
+            // Also, the contracts expect a property named 'instanceHandle'
+            // whereas our tables name the primary key just 'handle'.
+            return harden({
+              ...instanceRecord,
+              instanceHandle,
+            });
+          });
+      };
+
+      // The assays may not have been seen before, so we must wait for
+      // the assay records to be available synchronously
+      return assayTable
+        .getPromiseForAssayRecords(terms.assays)
+        .then(makeContractInstance)
+        .then(storeContractInstance);
     },
     /**
-     * Credibly retrieves an instance given an instanceHandle.
-     * @param {object} instanceHandle - the unique object for the instance
+     * Credibly retrieves an instance record given an instanceHandle.
+     * @param {object} instanceHandle - the unique, unforgeable
+     * identifier (empty object) for the instance
      */
-    getInstance: instanceHandle => {
-      const instance = adminState.getInstance(instanceHandle);
-      const installationHandle = adminState.getInstallationHandleForInstanceHandle(
-        instanceHandle,
-      );
-      const terms = readOnlyState.getTerms(instanceHandle);
-      return harden({
-        instanceHandle,
-        instance,
-        installationHandle,
-        terms,
-      });
-    },
+    getInstance: instanceTable.get,
 
     /**
-     * @param  {payoutRule[]} payoutRules - the offer description, an
-     * array of objects with `kind` and `units` properties.
+     * Escrow with Zoe and receive an escrow receipt and payout
+     * promise.
+     * @param  {offerRule[]} offerRules - the offer rules, an object
+     * with properties `payoutRules` and `exitRule`.
      * @param  {payment[]} offerPayments - payments corresponding to
-     * the offer description. A payment may be `undefined` in the case
-     * of specifying a `want`.
+     * the offer rules. A payment may be `undefined` in the case of
+     * specifying a `want`.
      */
-    escrow: async (offerRules, offerPayments) => {
-      const { offerHandle, result } = await escrowOffer(
-        adminState.recordOffer,
-        adminState.recordAssay,
-        offerRules,
-        offerPayments,
-      );
+    escrow: (offerRules, offerPayments) => {
+      const assays = getAssaysFromPayoutRules(offerRules.payoutRules);
+      const offerImmutableRecord = {
+        instanceHandle: undefined,
+        payoutRules: offerRules.payoutRules,
+        exitRule: offerRules.exitRule,
+        assays,
+        units: undefined,
+        extents: undefined,
+      };
 
-      const escrowReceiptPaymentP = mintEscrowReceiptPayment(
-        escrowReceiptMint,
-        offerHandle,
-        offerRules,
-      );
+      // units should only be gotten after the payments are deposited
+      const offerHandle = offerTable.create(offerImmutableRecord);
```

Comment:
See note on timing issue above. Assuming most of this logic survives the upcoming receipt removal, please let's try to consolidate most of this escrowing logic somehow, even in this PR, because it's too hard to review for equivalence across changes.

OTOH, if the upcoming receipt removal really makes more of this code irrelevant, and if it is hard to consolidate, then don't.